### ISSUE TITLE
Implement queueTask() and related functions

### ISSUE THREAD

**Issue by GreatEmerald** [Open-EO/openeo-r-client#13]
Some endpoints currently only support async calls to the `/jobs` endpoint (R backend included). There should be support for sending those jobs via the `queueTask()` function. `cancelJob()` would also be useful to help those backends clear up job descriptors. And then `orderResult()` is also just a variant of `queueTask()` that requests a download (link?).

**Comment by flahn** [comment]
I have been working on this already last weekend. In the development branch should be something that follows API v0.0.1.

I have not implemented `cancelJob() `, but `deleteJob` which deletes the job on the backend. Canceling would simply mean: "leave it on the server, but don't allow any processing on it". 

From a technical perspective `orderResult` and `queueTask` will communicate via the same client function (`con$uploadJob`). Since we defined it this way, I sticked to the naming (for now).
